### PR TITLE
More ubuntu fixes...

### PR DESCRIPTION
**Pull request by diranged**
@chaochenq, this PR reduces a bit of the complexity around babysitting the agent process, and fixes a few Ubuntu issues we discovered as we started implementing the agent.

### CODE REVIEW

**Comment by chaochenq** [comment]
Hey @diranged the main reason why we had a separate babysitter script was that we wanted to make sure that the agent process wasn't killed on purpose. That being said, right now if the agent was down by running `sudo service aws-kinesis-agent stop`, then `sudo service aws-kinesis-agent status` would return status code of 3 saying it's stopped, in which case we don't want the agent to be started automatically. If the agent was accidentally killed for some reason, `sudo service aws-kinesis-agent status` would return 1 saying "aws-kinesis-agent dead but pid file exists". And this is the case that babysitter wants to catch. So looks like this PR treats the two situation in the same manner?

File: bin/aws-kinesis-agent.RedHat [hunk 1]
```diff
@@ -151,6 +151,7 @@ do_stop () {
 do_status () {
   status -p $PIDFILE $DAEMON_NAME
   RETVAL=$?
+  exit $RETVAL
```
Comment (chaochenq) [inline bin/aws-kinesis-agent.RedHat]: The exit code will be returned at the end of the script anyway

File: support/aws-kinesis-agent.cron [hunk 2]
```diff
@@ -1 +1 @@
-* * * * * root /usr/bin/aws-kinesis-agent-babysit > /dev/null 2>&1
+* * * * * root service aws-kinesis-agent status > /dev/null 2>&1 || service aws-kinesis-agent start
```
Comment (chaochenq) [inline support/aws-kinesis-agent.cron]: If we want to make the babysitter script less complex, we want to at least differentiate whether the agent was intent to be brought down or not

File: bin/aws-kinesis-agent.RedHat (hunk 1 above)
Comment (diranged) [inline bin/aws-kinesis-agent.RedHat]: Duh good point

**Comment by diranged** [comment]
@chaochenq  indeed I misunderstood the point of the babysit script -- I've added it back in..

File: bin/aws-kinesis-agent-babysit [hunk 3]
```diff
@@ -2,9 +2,10 @@
 
 # The cron-babysit periodically checks the status of the agent, and restart it if stopped abnormally
 
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
```
Comment (chaochenq) [inline bin/aws-kinesis-agent-babysit]: Shall we do PATH=/sbin:/usr/sbin:/bin:/usr/bin:$PATH in case we don't want it to be overridden?

Comment (diranged) [inline bin/aws-kinesis-agent-babysit]: Done!

**Pull request merged**